crates/rpc/src/gas_price.rs

```diff
@@ -61,11 +61,7 @@ impl Cached {
                 // Update the gas price from the starknet pending block.
                 tokio::spawn(async move {
                     use starknet_gateway_types::reply::MaybePendingBlock;
-                    let gas_price = match gateway
-                        // Don't indefinitely retry as this could block the RPC request.
-                        .block_without_retry(pathfinder_common::BlockId::Pending)
-                        .await
-                    {
+                    let gas_price = match gateway.block(pathfinder_common::BlockId::Pending).await {
```

Comment:
My concern with moving the retry decision out from here is that this information is no longer local. The outside main caller has no idea how the gateway is being used in this context.

i.e. the decision to never retry in this exact case should be done locally?

Reply:
+1